src/server/scripts/Battlegrounds/DeephaulRavine/battleground_deephaul_ravine.cpp

```diff
@@ -0,0 +1,1094 @@
+        static constexpr uint32 ControlVisualNeutral = 430178;
+        static constexpr uint32 ControlVisualAlliance = 430221;
+
+        static constexpr uint32 CartControlCapturePointUnitEast = 430211;
+        static constexpr uint32 CartControlCapturePointUnitWest = 430216;
+
+        static constexpr uint32 CartCap = 430207;
+
+        static constexpr uint32 DefendingCartAura = 128646;
+
+        static constexpr uint32 Restoration = 296575;
+        static constexpr uint32 Berserking = 296576;
+        static constexpr uint32 RuneOfFrequency = 422968;
+
+        static constexpr uint32 CartExhaustion = 456466;
+    }
+
+    namespace WorldStates
+    {
+        static constexpr int32 FlagEnabled = 25412; // when this is 3, the flag is spawned
+
+        static constexpr int32 HordeTeamScore = 24959;
+        static constexpr int32 AllianceTeamScore = 24958;
+
+        static constexpr int32 MaxTeamScore = 24960;
+
+        static constexpr int32 BattleBegun = 7852;
+
+        static constexpr int32 AllianceControlsEastMineCart = 25415;
+        static constexpr int32 HordeControlsEastMineCart = 25414;
+
+        static constexpr int32 AllianceControlsWestMineCart = 25421;
+        static constexpr int32 HordeControlsWestMineCart = 25420;
+
+        static constexpr std::array<int32, 2> AllianceControlWorldStates = { AllianceControlsEastMineCart, AllianceControlsWestMineCart };
+        static constexpr std::array<int32, 2> HordeControlWorldStates = { HordeControlsEastMineCart, HordeControlsWestMineCart };
+
+        /*
+         * 1; on reset; dropped
+         * 2; when taken
+         */
+        static constexpr int32 HordeFlagState = 1546;
+        /*
+         * 1; on reset; dropped
+         * 2; when taken
+         */
+        static constexpr int32 AllianceFlagState = 1545;
+
+        // WSE: 12385, "(WorldState(6954) + WorldState(6955)) == 1")
+        // values might be swapped, they are not send to client
+        // See SilvershardMines
+        static constexpr int32 AllianceCapturedCart = 6954;
+        static constexpr int32 HordeCapturedCart = 6955;
+    }
+}
+
+struct battleground_deephaul_ravine : BattlegroundScript
+{
+    explicit battleground_deephaul_ravine(BattlegroundMap* map) : BattlegroundScript(map), _cartsReached(0), _cartGUIDs({ }), _capturePointAreaTriggers({ }), _leaderGUIDs({ }) { }
+
+    void OnInit() override
+    {
+        BattlegroundScript::OnInit();
+
+        for (Position const& pos : DeephaulRavine::Positions::EarthenMineCartsHorde)
+            battlegroundMap->SummonCreature(DeephaulRavine::Creatures::EarthenMineCartHorde, pos);
+
+        for (Position const& pos : DeephaulRavine::Positions::EarthenMineCartsAlliance)
+            battlegroundMap->SummonCreature(DeephaulRavine::Creatures::EarthenMineCartAlliance, pos);
+    }
+
+    void OnStart() override
+    {
+        BattlegroundScript::OnStart();
+        _scheduler.Schedule(15s, [&](TaskContext)
```

Comment:
Small improvement: do you think you can move any timer to its own enum like Timings? You can make them static constexpr std::chrono::seconds, so it's easier to maintain if anything changes rather than having magic numbs.